model/pdata/common.go

```diff
@@ -525,6 +525,26 @@ func (am AttributeMap) Delete(key string) bool {
 	return false
 }
 
+// Delete deletes the entry for which the function in question
+func (am AttributeMap) DeleteIf(f func(string, AttributeValue) bool) {
```

Comment:
For consistency, rename this to "RemoveIf" since that is what we use in slices? It is just a suggestion, opposite argument can be made since we call "Delete" not "Remove" the other API. Maybe change of fixing that as well.

Reply:
Changed DeleteIf -> RemoveIf and Delete -> Remove.  I added a deprecation warning to Delete, and made it call Remove